File: src/bmsx/core/host_asset_sync.ts
import type { Memory } from '../machine/memory/memory';
import { syncDirtyRuntimeImageAssets } from '../render/runtime_asset_textures';
import type { TextureManager } from '../render/texture_manager';

export function flushHostRuntimeAssetEdits(memory: Memory, texmanager: TextureManager): void {
	const dirtyAssets = memory.consumeDirtyAssets();
	if (dirtyAssets.length === 0) {
		return;
	}
	syncDirtyRuntimeImageAssets(memory, dirtyAssets, texmanager);
}

File: src/bmsx_cpp/core/host_asset_sync.cpp
#include "core/host_asset_sync.h"

#include "machine/memory/memory.h"
#include "render/gameview.h"
#include "render/runtime_asset_textures.h"

namespace bmsx {

void flushHostRuntimeAssetEdits(Memory& memory, TextureManager& texmanager, const GameView& view) {
	if (!view.backend()->readyForTextureUpload()) {
		return;
	}
	auto dirtyAssets = memory.consumeDirtyAssets();
	if (dirtyAssets.empty()) {
		return;
	}
	RuntimeAssetTextures::syncDirtyRuntimeImageAssets(memory, dirtyAssets, texmanager);
}

} // namespace bmsx
